dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/DatadogCoreProxy.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core
+
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureScope
+import com.datadog.android.api.storage.EventBatchWriter
+import com.datadog.android.api.storage.RawBatchEvent
+import com.google.gson.Gson
+import com.google.gson.JsonElement
+
+class DatadogCoreProxy(
```

Comment:
I think this is very similar to https://github.com/DataDog/dd-sdk-android/blob/develop/reliability/stub-core/src/main/kotlin/com/datadog/android/core/stub/StubSDKCore.kt we have (but it is not published).

If this one is needed for the testing, maybe we can simply inject it into the necessary features without registering in the registry? In fact, very few components rely on the registry internally (like OkHttp instrumentation, which needs to capture the core in the lazy manner, hence it is using name -> registry), and probably none of them are used by the RN.

Reply:
I think it would make sense to rely on the existing Stub we are using internally, instead of duplicating the same feature. In addition, it'd be better to keep test purpose classes in a test artifact that would not be shipped in an app

Reply:
I believe this is the most important comment so I'll focus on this one before jumping to the others :) 

The goal of this Core Proxy is to be able to run assertions on events that were received by the core in the context of a real app (for instance to test React Native Session Replay). 

Also as I am using the RN SDK in the testing app I cannot interfere too much on how the SDK is going to be initialized except through config fields.
I can potentially change some gradle files but I'd like to avoid having to touch actual code of the RN SDK.

Maybe there is a way to release this as a separate artifact that could be used by the RN SDK without making too many changes to it, if so let me know, I can take time to explore more solutions :) 

Let me know if this is clearer and if you see a better alternative here.